lib/teleterm/apiserver/apiserver.go

```diff
@@ -92,3 +111,58 @@ type APIServer struct {
 	// grpc is an instance of grpc server
 	grpcServer *grpc.Server
 }
+
+func getGrpcCredentials(cfg Config) (grpc.ServerOption, error) {
+	uri, err := utils.ParseAddr(cfg.HostAddr)
+
+	if err != nil {
+		return nil, trace.BadParameter("invalid host address: %s", cfg.HostAddr)
+	}
+
+	if uri.Network() != "unix" {
+		keyPair, err := generateKeyPair(cfg.CertsDir)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		return grpc.Creds(keyPair), nil
+	}
+
+	return grpc.Creds(nil), nil
+}
+
+func generateKeyPair(certsDir string) (credentials.TransportCredentials, error) {
+	cert, err := utils.GenerateSelfSignedCert([]string{"localhost"})
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to generate a certificate")
+	}
+
+	err = os.WriteFile(filepath.Join(certsDir, "tsh_server.crt"), cert.Cert, 0600)
```

Comment:
This may very well leave the `tsh_server.crt` file half written for long enough that Connect can read it and complain about a malformed certificate. I don't think we need to find some cross-platform atomic file renamer (as `os.Rename()` is not actually atomic on windows) but maybe deleting any preexisting `tsh_server.crt`, writing the certificate in `tsh_server.tmp` and then renaming is going to be enough?

Reply:
Good catch, I pushed a fix for that. 
The certs dir is cleared when the app starts, so we don't have to remove the file in `tshd`.